internal/identity/accumulator.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2021 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package identity
+
+import (
+	"github.com/twmb/murmur3"
+)
+
+const (
+	_hashSeed uint64 = 23
+	_hashFold uint64 = 31
+)
+
+// Accumulator is a commutative folding accumulator.
+type Accumulator uint64
+
+// NewAccumulator creates a new Accumulator with a default seed value.
+//
+// n.b. Here and elsewhere, we use nosplit to avoid stack size checks, which
```

Comment:
thanks!

Reply:
I'm a little cautious when using the Go pragmas (seems a little more "internal"). Do benchmarks show that these provide provide a significant enough improvement to justify these? 

I'd expect most of these simple methods would be inlined anyway, so `go:nosplit` likely won't have any impact,
```
./accumulator.go:43:6: can inline NewAccumulator
./accumulator.go:49:6: can inline NewAccumulatorWithSeed
./accumulator.go:55:6: can inline Accumulator.AddString
./accumulator.go:71:6: can inline Accumulator.AddUint64
./accumulator.go:87:6: can inline Accumulator.Value
```

Reply:
Fair point - I'd been looking at these in a vacuum. It's been a bit since I had done benchmarks on this but I don't think there's a lot of value in hinting about stack checks when they'll be inlined. Thanks!